qr-code/src/main/java/org/odk/collect/qrcode/mlkit/MlKitBarcodeScannerViewFactory.kt

```diff
@@ -102,12 +102,15 @@ private class MlKitBarcodeScannerView(
                 val value = result.getValue(barcodeScanner)
                 if (value!!.isNotEmpty()) {
                     callback(value.first().rawValue!!)
-                    cameraController.unbind()
                 }
             }
         )
     }
 
+    override fun finish() {
+        cameraController.unbind()
```

Comment:
I'm really confused by this as I would have expected that the earlier `bindToLifecycle` call would handle this clean up. Is there potentially a problem with the lifecycle we're passing in to `BarcodeScannerViewContainer#setup` in `QrCodeProjectCreatorDialog`? I'm now remembering there are problems with using `viewLifecycleOwner` within `DialogFragment` (https://developer.android.com/guide/fragments/dialogs).

Reply:
You added that in this commit https://github.com/getodk/collect/commit/979f704f706a58675d5aa8d9065dadc83b1f0c7d so I guess tried to solve something? Don't you remember?

Reply:
Ahh I thought it was added to handle some memory leaks or something... ok now I have a different solution: https://github.com/getodk/collect/pull/6752